runtimevar/runtimevar_test.go

```diff
@@ -507,21 +506,35 @@ func TestURLMux(t *testing.T) {
 			if gotErr != nil {
 				return
 			}
-			want, err := url.Parse(tc.url)
+			if got := fake.u.String(); got != tc.want {
+				t.Errorf("got %q want %q", got, tc.want)
+			}
+			// Repeat with OpenVariableURL.
+			parsed, err := url.Parse(tc.url)
 			if err != nil {
 				t.Fatal(err)
 			}
-			if diff := cmp.Diff(got, want); diff != "" {
-				t.Errorf("got\n%v\nwant\n%v\ndiff\n%s", got, want, diff)
+			_, gotErr = mux.OpenVariableURL(ctx, parsed)
+			if (gotErr != nil) != tc.wantErr {
+				t.Fatalf("got err %v, want error %v", gotErr, tc.wantErr)
+			}
```

Comment:
continue when gotErr != nil

Reply:
Actually, there shouldn't be any error here. If there was an expected error, we returned above. I cleaned up this code (here and elsewhere) to fail if there's an error.